man-page/uu_split.1
.ie \n(.g .ds Aq \(aq
.el .ds Aq '
.TH uu_split 1  "uu_split 0.0.27" 
.SH NAME
uu_split \- Create output files containing consecutive or interleaved sections of input
.SH SYNOPSIS
\fBuu_split\fR [\fB\-b\fR|\fB\-\-bytes\fR] [\fB\-C\fR|\fB\-\-line\-bytes\fR] [\fB\-l\fR|\fB\-\-lines\fR] [\fB\-n\fR|\fB\-\-number\fR] [\fB\-\-additional\-suffix\fR] [\fB\-\-filter\fR] [\fB\-e\fR|\fB\-\-elide\-empty\-files\fR] [\fB\-d \fR] [\fB\-\-numeric\-suffixes\fR] [\fB\-x \fR] [\fB\-\-hex\-suffixes\fR] [\fB\-a\fR|\fB\-\-suffix\-length\fR] [\fB\-\-verbose\fR] [\fB\-t\fR|\fB\-\-separator\fR] [\fB\-h\fR|\fB\-\-help\fR] [\fB\-V\fR|\fB\-\-version\fR] [\fIinput\fR] [\fIprefix\fR] 
.SH DESCRIPTION
Create output files containing consecutive or interleaved sections of input
.SH OPTIONS
.TP
\fB\-b\fR, \fB\-\-bytes\fR=\fISIZE\fR
put SIZE bytes per output file
.TP
\fB\-C\fR, \fB\-\-line\-bytes\fR=\fISIZE\fR
put at most SIZE bytes of lines per output file
.TP
\fB\-l\fR, \fB\-\-lines\fR=\fINUMBER\fR [default: 1000]
put NUMBER lines/records per output file
.TP
\fB\-n\fR, \fB\-\-number\fR=\fICHUNKS\fR
generate CHUNKS output files; see explanation below
.TP
\fB\-\-additional\-suffix\fR=\fISUFFIX\fR [default: ]
additional SUFFIX to append to output file names
.TP
\fB\-\-filter\fR=\fICOMMAND\fR
write to shell COMMAND; file name is $FILE (Currently not implemented for Windows)
.TP
\fB\-e\fR, \fB\-\-elide\-empty\-files\fR
do not generate empty output files with \*(Aq\-n\*(Aq
.TP
\fB\-d\fR
use numeric suffixes starting at 0, not alphabetic
.TP
\fB\-\-numeric\-suffixes\fR=\fIFROM\fR
same as \-d, but allow setting the start value
.TP
\fB\-x\fR
use hex suffixes starting at 0, not alphabetic
.TP
\fB\-\-hex\-suffixes\fR=\fIFROM\fR
same as \-x, but allow setting the start value
.TP
\fB\-a\fR, \fB\-\-suffix\-length\fR=\fIN\fR
generate suffixes of length N (default 2)
.TP
\fB\-\-verbose\fR
print a diagnostic just before each output file is opened
.TP
\fB\-t\fR, \fB\-\-separator\fR=\fISEP\fR
use SEP instead of newline as the record separator; \*(Aq\\0\*(Aq (zero) specifies the NUL character
.TP
\fB\-h\fR, \fB\-\-help\fR
Print help
.TP
\fB\-V\fR, \fB\-\-version\fR
Print version
.TP
[\fIinput\fR] [default: \-]

.TP
[\fIprefix\fR] [default: x]

.SH EXTRA
Output fixed\-size pieces of INPUT to PREFIXaa, PREFIXab, ...; default size is 1000, and default PREFIX is \*(Aqx\*(Aq. With no INPUT, or when INPUT is \-, read standard input.

The SIZE argument is an integer and optional unit (example: 10K is 10*1024).
Units are K,M,G,T,P,E,Z,Y,R,Q (powers of 1024) or KB,MB,... (powers of 1000).
Binary prefixes can be used, too: KiB=K, MiB=M, and so on.

CHUNKS may be:

\- N       split into N files based on size of input
\- K/N     output Kth of N to stdout
\- l/N     split into N files without splitting lines/records
\- l/K/N   output Kth of N to stdout without splitting lines/records
\- r/N     like \*(Aql\*(Aq but use round robin distribution
\- r/K/N   likewise but only output Kth of N to stdout
.SH VERSION
v0.0.27


completion/uu_split.bash
_uu_split() {
    local i cur prev opts cmd
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"
    cmd=""
    opts=""

    for i in ${COMP_WORDS[@]}
    do
        case "${cmd},${i}" in
            ",$1")
                cmd="uu_split"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        uu_split)
            opts="-b -C -l -n -e -d -x -a -t -h -V --bytes --line-bytes --lines --number --additional-suffix --filter --elide-empty-files --numeric-suffixes --hex-suffixes --suffix-length --verbose --separator --io-blksize --help --version [input] [prefix]"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --bytes)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -b)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --line-bytes)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -C)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --lines)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -l)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --number)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -n)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --additional-suffix)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --filter)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --numeric-suffixes)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --hex-suffixes)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --suffix-length)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -a)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --separator)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -t)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --io-blksize)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

complete -F _uu_split -o nosort -o bashdefault -o default uu_split
